packages/component-test-utils-react/src/shallow.js

```diff
@@ -2,45 +2,61 @@ const React = require('react');
 const {getHtml} = require('./methods/html');
 const {shallowMock} = require('./shallow-mock');
 const {createDispatcher} = require('./dispatcher/');
+const {dispatchEvent} = require('./methods/dispatchEvent');
 
 const {
   ReactCurrentDispatcher
 } = React.__SECRET_INTERNALS_DO_NOT_USE_OR_YOU_WILL_BE_FIRED;
 
 class ShallowRender {
-  constructor() {
+  constructor(component, config = {}) {
+    if (!component) {
+      return;
+    }
+
+    this._component = component;
+    this._config = config;
     this._context = null;
-    this._dispatcher = createDispatcher();
+    this._dispatcher = createDispatcher(this);
+
+    this._render();
+
+    // Finish recording the order of hooks by toogling this dispatcher property
+    this._dispatcher._firstCall = false;
+
+    // Render mocked component
+    if (this._config.mock && this._rendered.props.children) {
+      this._rendered = shallowMock(
+        this._rendered,
+        this._config.mock,
+        ShallowRender
+      );
+    }
   }
 
-  shallow(reactElement, config = {}) {
+  _render() {
     const prevDispatcher = ReactCurrentDispatcher.current;
     ReactCurrentDispatcher.current = this._dispatcher;
 
-    this._currentlyRenderingComponent = reactElement.type;
-    this._previousComponentIdentity = reactElement.type;
-
-    this._rendered = reactElement.type.call(
+    this._rendered = this._component.type.call(
       undefined,
-      reactElement.props,
+      this._component.props,
       this._context
     );
 
-    // Render mocked component
-    if (config.mock && this._rendered.props.children) {
-      this._rendered = shallowMock(this._rendered, config.mock, ShallowRender);
-    }
-
     ReactCurrentDispatcher.current = prevDispatcher;
+  }
+
+  // Methods
+  html() {
+    return getHtml(this._rendered);
+  }
 
-    return {
-      html: () => {
-        return getHtml(this._rendered);
-      }
-    };
+  dispatchEvent(event) {
+    return dispatchEvent(this._rendered, event);
   }
 }
 
 exports.shallow = (component, config) => {
-  return new ShallowRender().shallow(component, config);
+  return new ShallowRender(component, config);
```

Comment:
Or make the validation here so you don't create an invalid ShallowRenderer ?

Reply:
The validation will be there and will be the objet of a future PR ;)

The package component-test-utils-vuejs is already full of beautiful error messages I need to share here